packages/@uppy/utils/src/getDroppedFiles/utils/webkitGetAsEntryApi/index.js

```diff
@@ -25,19 +25,19 @@ function getAsFileSystemHandleFromEntry (entry, logDropError) {
   }
 }
 
-async function* createPromiseToAddFileOrParseDirectory (entry, relativePath) {
+async function* createPromiseToAddFileOrParseDirectory (entry, relativePath, lastResortFile = undefined) {
   // For each dropped item, - make sure it's a file/directory, and start deepening in!
   if (entry.kind === 'file') {
     const file = await entry.getFile()
     if (file !== null) {
       file.relativePath = relativePath ? `${relativePath}/${entry.name}` : null
       yield file
-    }
+    } else if (lastResortFile != null) yield lastResortFile
   } else if (entry.kind === 'directory') {
     for await (const handle of entry.values()) {
       yield* createPromiseToAddFileOrParseDirectory(handle, `${relativePath}/${entry.name}`)
```

Comment:
Why are we using generator and `yield` here? Not saying we shouldn't, but it’s hard to read and I think this is the only place we use them.

Reply:
"hard to read" is subjective, I kinda doubt this would be more readable without using generators – not saying that this code is the most readable, but rather than we could (should?) make it way more readable while keeping generators. In my experience using async generators is the "best" way to iterate through a file tree, as in it's probably the most memory efficient while being the more elegant solution.